### PR TITLE
Add ability to communicate over unix domain sockets

### PR DESCRIPTION
**Pull request by imurchie**

### CODE REVIEW

File: lib/remote-debugger-rpc-client.js [hunk 1]
```diff
@@ -14,11 +14,15 @@ import { simpleStringify } from './helpers';
 
 
 export default class RemoteDebuggerRpcClient {
-  constructor (host, port = REMOTE_DEBUGGER_PORT, specialMessageHandlers = {}) {
-    // socket and connection configuration
+  constructor (opts = {}) {
+    let {host, port, socketPath, specialMessageHandlers} = opts;
```
Comment (mykola-mokhnach) [inline lib/remote-debugger-rpc-client.js]: you can set default values right here

File: lib/remote-debugger-rpc-client.js [hunk 2]
```diff
@@ -14,11 +14,15 @@ import { simpleStringify } from './helpers';
 
 
 export default class RemoteDebuggerRpcClient {
-  constructor (host, port = REMOTE_DEBUGGER_PORT, specialMessageHandlers = {}) {
-    // socket and connection configuration
+  constructor (opts = {}) {
```
Comment (mykola-mokhnach) [inline lib/remote-debugger-rpc-client.js]: do we need to raise the major lib version after his change?

Comment (imurchie) [reply lib/remote-debugger-rpc-client.js]: No. Everything works exactly as it did unless you pass in the socket path, at which time things change.

File: lib/remote-debugger.js [hunk 3]
```diff
@@ -39,22 +39,32 @@ class RemoteDebugger extends events.EventEmitter {
   constructor (opts = {}) {
     super();
 
-    let {bundleId, platformVersion, debuggerType, useNewSafari, pageLoadMs,
-         host, port, pageReadyTimeout} = opts;
+    const {
```
Comment (jlipps) [inline lib/remote-debugger.js]: ❤️ the way this pattern looks

**Pull request merged**